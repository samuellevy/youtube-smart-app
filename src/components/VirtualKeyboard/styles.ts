import styled from 'styled-components';
import { colors } from '../../global';

interface IKey {
  active?:boolean;
}

export const Container = styled.div`
  background-color: #000000aa;
  width: 600px;
  display: flex;
  flex-direction: column;
`;

export const Line = styled.div`
  display: flex;
  justify-content: center;
`;

export const Key = styled.div<IKey>`
  border: 1px black solid;
  padding: 10px 15px;
  background-color: ${(props) => (props.active ? colors.darkGray : colors.grayBg)}
`;
