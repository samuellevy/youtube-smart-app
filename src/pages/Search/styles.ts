import styled from 'styled-components';
import { colors } from '../../global';

interface IContainer {
  active?: boolean;
}

export const Container = styled.div<IContainer>`
  padding: 25px 0 0 25px;
  padding-left: 250px;
  transform: ${(props) => (props.active ? 'translateX(-140px)' : 'translateX(0)')};
  transition: transform 200ms linear;
`;
export const PageTitle = styled.h1`
  color: ${colors.gray};
  font-size: 1vw;
  text-transform: uppercase;
`;
export const Track = styled.div`
  display: flex;
`;
export const CategoryTitle = styled.h1`
  width: 100%;
`;
