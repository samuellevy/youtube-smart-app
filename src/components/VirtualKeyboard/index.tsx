import React, { useEffect, useState } from 'react';
import { useKeyboardContext } from '../../context/KeyboardContext';

import * as K from './styles';

interface IVirtualKeyboard{
  handleQuery(qs: string): void;
}

const VirtualKeyboard: React.FC<IVirtualKeyboard> = ({ handleQuery }: IVirtualKeyboard) => {
  const keyMap = [
    ['1', '2', '3', '4', '5', '6', '7', '8', '9', '0', '-', '+', 'del'],
    ['q', 'w', 'e', 'r', 't', 'y', 'u', 'i', 'o', 'p', '[', ']', '\\'],
    ['a', 's', 'd', 'f', 'g', 'h', 'j', 'k', 'l', ';', '"', 'enter'],
    ['shift', 'z', 'x', 'c', 'v', 'b', 'n', 'm', ',', '.', '?'],
    ['space'],
    [],
  ];

  const { keyboard, dispatch }: any = useKeyboardContext();
  const [activeKeyboard, setActiveKeyboard] = useState(true);
  const [activeKey, setActiveKey] = useState([-1, -1]);
  const [search, setSearch] = useState('');

  useEffect(() => {
    if (activeKeyboard && keyboard.component === 'search') {
      if (activeKey[0] !== -1 && activeKey[0] !== -1) {
        controlHandler(keyboard.key);
      } else {
        setActiveKey([0, 0]);
      }
    }
  }, [keyboard]);

  const controlHandler = (key: string) => {
    let newKeyX = activeKey[0];
    let newKeyY = activeKey[1];

    switch (key) {
      case 'ArrowRight':
        newKeyX += newKeyX + 1 < keyMap[newKeyY].length ? 1 : 0;
        newKeyY += 0;
        break;
      case 'ArrowLeft':
        newKeyX -= newKeyX > 0 ? 1 : 0;
        newKeyY += 0;
        break;
      case 'ArrowDown':
        newKeyY += newKeyY + 2 < keyMap.length ? 1 : 0;
        newKeyX = keyMap[newKeyY + 1]?.length < newKeyX
          ? keyMap[newKeyY].length - 1 : newKeyX + 0;
        break;
      case 'ArrowUp':
        newKeyX += 0;
        newKeyY -= newKeyY > 0 ? 1 : 0;
        break;
      case 'Enter':
        handleQuery(keyMap[activeKey[1]][activeKey[0]]);
        break;
      default:
        break;
    }
    setActiveKey([newKeyX, newKeyY]);
  };

  // const handleQuery = (qs: string) => {
  //   console.log(qs);
  // };

  return (
    <K.Container>
      {keyMap.map((column: any, keyY: number) => (
        <K.Line>
          {column.map((key: any, keyX: number) => (
            <K.Key active={activeKey[0] === keyX && activeKey[1] === keyY}>{key}</K.Key>
          ))}
        </K.Line>
      ))}
    </K.Container>
  );
};

export default VirtualKeyboard;
