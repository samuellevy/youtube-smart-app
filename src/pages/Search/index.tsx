import React, { useEffect, useState } from 'react';
import Track from '../../components/Track';
import { useKeyboardContext, changeComponent } from '../../context/KeyboardContext';
// import VideoCard from '../../components/VideoCard';

import * as S from './styles';

const Search: React.FC = () => {
  const { keyboard, dispatch }: any = useKeyboardContext();
  const [activePage, setActivePage] = useState(false);

  useEffect(() => {
    dispatch(changeComponent('menu'));
  }, []);

  useEffect(() => {
    if (keyboard.component === 'search') {
      setActivePage(true);
    }
  }, [keyboard]);

  return (
    <S.Container active={activePage}>
      <S.PageTitle>Search</S.PageTitle>
    </S.Container>
  );
};
export default Search;
