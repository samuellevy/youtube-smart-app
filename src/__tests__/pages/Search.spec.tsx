import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Search from '../../pages/Search';
import VirtualKeyboard from '../../components/VirtualKeyboard';

Enzyme.configure({ adapter: new Adapter() });

describe('<Search /> component', () => {
  it('should render properly', () => {
    const tree = renderer.create(<Search />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render VirtualKeyboard component', () => {
    const tree = shallow(<Search />).dive();
    expect(tree.exists('VirtualKeyboard')).toBeTruthy();
  });
});
