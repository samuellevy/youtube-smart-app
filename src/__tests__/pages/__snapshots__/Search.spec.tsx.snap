// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> component should render properly 1`] = `
<div
  className="sc-hKgILt dhHBuA"
>
  <h1
    className="sc-eCssSg iFyiDB"
  >
    
  </h1>
  <div
    className="sc-bdfBwQ kVOLiH"
  >
    <div
      className="sc-gsTCUz kNNvzq"
    >
      <div
        className="sc-dlfnbm ghevfs"
      >
        1
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        2
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        3
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        4
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        5
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        6
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        7
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        8
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        9
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        0
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        -
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        +
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        del
      </div>
    </div>
    <div
      className="sc-gsTCUz kNNvzq"
    >
      <div
        className="sc-dlfnbm ghevfs"
      >
        q
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        w
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        e
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        r
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        t
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        y
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        u
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        i
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        o
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        p
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        [
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        ]
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        \\
      </div>
    </div>
    <div
      className="sc-gsTCUz kNNvzq"
    >
      <div
        className="sc-dlfnbm ghevfs"
      >
        a
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        s
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        d
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        f
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        g
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        h
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        j
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        k
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        l
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        ;
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        "
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        enter
      </div>
    </div>
    <div
      className="sc-gsTCUz kNNvzq"
    >
      <div
        className="sc-dlfnbm ghevfs"
      >
        shift
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        z
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        x
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        c
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        v
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        b
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        n
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        m
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        ,
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        .
      </div>
      <div
        className="sc-dlfnbm ghevfs"
      >
        ?
      </div>
    </div>
    <div
      className="sc-gsTCUz kNNvzq"
    >
      <div
        className="sc-dlfnbm ghevfs"
      >
        space
      </div>
    </div>
    <div
      className="sc-gsTCUz kNNvzq"
    />
  </div>
</div>
`;
